launcher/src/main/java/org/apache/spark/launcher/SparkSubmitCommandBuilder.java

```diff
@@ -392,7 +392,7 @@ boolean isClientMode(Map<String, String> userProps) {
     // Default master is "local[*]", so assume client mode in that case
     return userMaster == null ||
       "client".equals(userDeployMode) ||
-      (!userMaster.equals("yarn") && userDeployMode == null);
+      userDeployMode == null;
```

Comment:
In addition, could you add a test case to prevent a future regression, @sandeep-katta ?

Reply:
ya ll add testcase and update the PR